include/swift/AST/ASTContext.h

```diff
@@ -521,6 +521,18 @@ class ASTContext final {
                               setVector.size());
   }
 
+  template <typename Output, typename Range>
+  ArrayRef<Output> AllocateTransform(
+      Range &&input,
+      llvm::function_ref<Output(const decltype(input.front()) &)> transform,
```

Comment:
Is the transform param type intentionally not `const Range::value_type &` or `Range::const_reference`?